src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -22,42 +23,99 @@ public DefaultClaimUidExtractor(ObjectPool<AntiforgerySerializationContext> pool
         }
 
         /// <inheritdoc />
-        public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
+        public string ExtractClaimUid(ClaimsPrincipal claimsPrincipal)
         {
-            if (claimsIdentity == null || !claimsIdentity.IsAuthenticated)
+            Debug.Assert(claimsPrincipal != null);
+
+            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsPrincipal.Identities);
+            if (uniqueIdentifierParameters == null)
             {
                 // Skip anonymous users
                 return null;
             }
 
-            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsIdentity);
             var claimUidBytes = ComputeSha256(uniqueIdentifierParameters);
             return Convert.ToBase64String(claimUidBytes);
         }
 
-        // Internal for testing
-        internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity claimsIdentity)
+        public static IList<string> GetUniqueIdentifierParameters(IEnumerable<ClaimsIdentity> claimsIdentities)
         {
-            var nameIdentifierClaim = claimsIdentity.FindFirst(
-                claim => string.Equals(ClaimTypes.NameIdentifier, claim.Type, StringComparison.Ordinal));
-            if (nameIdentifierClaim != null && !string.IsNullOrEmpty(nameIdentifierClaim.Value))
+            var identitiesList = claimsIdentities as List<ClaimsIdentity>;
+            if (identitiesList == null)
+            {
+                identitiesList = claimsIdentities.ToList();
```

Comment:
FYI `enumerable.ToList()` is pretty much identical to `new List<T>(enumerable)`. No need to change unless it allows you to remove the `using` for Linq; then would be good.